Source/ASCollectionView.mm

```diff
@@ -1131,37 +1143,45 @@ - (void)collectionView:(UICollectionView *)collectionView didEndDisplayingCell:(
 
 - (void)collectionView:(UICollectionView *)collectionView willDisplaySupplementaryView:(UICollectionReusableView *)rawView forElementKind:(NSString *)elementKind atIndexPath:(NSIndexPath *)indexPath
 {
-  if (rawView.class != [_ASCollectionReusableView class]) {
+  ASCollectionReusableViewCastOrReturn(rawView, view, (void)0);
```

Comment:
It turns out that we can't track visibility of elements that are not backed by a non-_ASCollection* view.  That is because in `-didEndDisplaying*` methods, we can't retrieve the element from visible map since the map might have been updated and no longer holds the element. So we have to rely on `cell.element`, otherwise we risk having imbalanced add and remove calls on `_visibleElements`, i.e elements that aren't backed by a non-_ASCollection* view will stay visible forever.